helios-services/src/main/java/com/spotify/helios/master/resources/HostsResource.java

```diff
@@ -147,13 +148,21 @@ public HostDeregisterResponse delete(@PathParam("id") final String host) {
   @Produces(APPLICATION_JSON)
   @Timed
   @ExceptionMetered
-  public Optional<HostStatus> hostStatus(@PathParam("id") final String host) {
-    return Optional.fromNullable(model.getHostStatus(host));
+  public Optional<HostStatus> hostStatus(@PathParam("id") final String host,
+                                         @QueryParam("status") final String statusFilter) {
```

Comment:
What happens in the status query-param is missing? It should still work, to remain backwards-compatible with older versions of the CLI - does it?

If it doesn't work without specifying the query-param you might want to use `@DefaultValue("")` or [guava] Optional.